File: 27.js
// Leetcode 27: Remove element
// Reference: https://leetcode.com/problems/remove-element/?envType=study-plan-v2&envId=top-interview-150

/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    var zeroStartIdx = 0;
    for (let i=0; i < nums.length; i++){
        if (nums[i] != val){
            nums[zeroStartIdx] = nums[i];
            zeroStartIdx++;
        }
    }
    return zeroStartIdx;
};

File: 27.java
// Leetcode 27: Remove element
// Reference: https://leetcode.com/problems/remove-element/solutions/3102906/java-best-solution-o-n-time-complexity/?envType=study-plan-v2&envId=top-interview-150

// Time Complexity: O(n)
class Solution {
    public int removeElement(int[] nums, int val) {
        int i = 0;
        for (int j = 0; j < nums.length; j++){
            if (nums[j] != val){
                int temp = nums[i];
                nums[i] = nums[j];
                nums[j] = temp;
                i++;
            }
        }
        return i;
    }
}


